File: app/src/androidTest/java/com/cottondroid/todoachieved/database/TestDatabaseModule.java
package com.cottondroid.todoachieved.database;

import android.arch.persistence.room.Room;

import com.cottondroid.todoachieved.TodoApplication;
import com.cottondroid.todoachieved.task.model.TodoTaskDao;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class TestDatabaseModule {

    @Provides
    @Singleton
    AppDatabase provideAppDatabase(TodoApplication application) {
        return Room.inMemoryDatabaseBuilder(application, AppDatabase.class).build();
    }

    @Provides
    @Singleton
    TodoTaskDao provideTodoTaskDao(AppDatabase appDatabase) {
        return appDatabase.todoTaskDao();
    }
}

File: app/src/androidTest/java/com/cottondroid/todoachieved/database/TestDatabaseModule.kt
package com.cottondroid.todoachieved.database

import androidx.room.Room
import com.cottondroid.todoachieved.TodoApplication
import com.cottondroid.todoachieved.task.model.TodoTaskDao
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class TestDatabaseModule {
    @Provides
    @Singleton
    fun provideAppDatabase(application: TodoApplication): AppDatabase {
        return Room.inMemoryDatabaseBuilder(application, AppDatabase::class.java)
                .allowMainThreadQueries()
                .build()
    }

    @Provides
    @Singleton
    fun provideTodoTaskDao(appDatabase: AppDatabase): TodoTaskDao {
        return appDatabase.todoTaskDao()
    }
}